pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.java

```diff
@@ -926,8 +933,12 @@ private static List<JobProperty> getPropertiesToUpdate(@CheckForNull List<JobPro
     @NonNull
     private static List<JobProperty> getPropertiesToRemove(@CheckForNull List<JobProperty> currentProperties,
                                                            @NonNull List<JobProperty> existingProperties) {
-        Set<String> currentPropertiesDescriptors = currentProperties.collect{ it.descriptor.id }.toSet()
-        return existingProperties.findAll{ !(it.descriptor.id in currentPropertiesDescriptors)}
+        Set<String> currentPropertiesDescriptors = currentProperties != null ? currentProperties.stream()
```

Comment:
Was able to resolve one SpotBugs violation here (`currentProperties` null check), but if I do the same in a nearby method, the diff goes over 50% modified: https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/765#discussion_r1970688710